src/java/org/apache/cassandra/utils/MBeanWrapper.java

```diff
@@ -170,6 +186,125 @@ public void unregisterMBean(ObjectName mbeanName, OnException onException)
         }
     }
 
+    static class InstanceMBeanWrapper implements MBeanWrapper
+    {
+        private final MBeanServer mbs;
+        public final UUID id = UUID.randomUUID();
+
+        public InstanceMBeanWrapper(String hostname)
+        {
+            mbs = MBeanServerFactory.createMBeanServer(hostname + "-" + id);
+        }
+
+        public void registerMBean(Object obj, ObjectName mbeanName, OnException onException)
+        {
+            try
+            {
+                mbs.registerMBean(obj, mbeanName);
+            }
+            catch (Exception e)
+            {
+                onException.handler.accept(e);
+            }
+        }
+
+        public boolean isRegistered(ObjectName mbeanName, OnException onException)
+        {
+            try
+            {
+                return mbs.isRegistered(mbeanName);
+            }
+            catch (Exception e)
+            {
+                onException.handler.accept(e);
+            }
+            return false;
+        }
+
+        public void unregisterMBean(ObjectName mbeanName, OnException onException)
+        {
+            try
+            {
+                mbs.unregisterMBean(mbeanName);
+            }
+            catch (Exception e)
+            {
+                onException.handler.accept(e);
+            }
+        }
+
+        public MBeanServer getMBeanServer()
+        {
+            return mbs;
+        }
+
+        public void close() {
+            mbs.queryNames(null, null).forEach(name -> {
+                try {
+                    if (!name.getCanonicalName().contains("MBeanServerDelegate"))
+                        mbs.unregisterMBean(name);
+                } catch (Throwable e) {
+                    logger.debug("Could not unregister mbean {}", name.getCanonicalName());
```

Comment:
log at warn maybe?
```suggestion
                    logger.warn("Could not unregister mbean {}", name.getCanonicalName(), e);
```

Reply:
The in-jvm dtest enables debug logging everywhere anyway, and I don't actually think this is a `warn`-level thing as the whole JMX infrastructure gets shut down eventually anyway.

Reply:
should we still log `e` here? or will it create too much output

Reply:
I don't think it's necessary - it generally happens because it's already been unregistered elsewhere (so it's actually not a huge problem, was more there for, well, debugging), and it would be super-verbose.